### PR TITLE
Mesh discovery

### PR DESCRIPTION
**Pull request by EItanya**

### CODE REVIEW

File: supergloo/docs/mesh/mesh-discovery.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+Mesh discovery, similar to other solo.io projects, uses an event loop based architecure to watch kubernetes 
+resources to create/update those resources. In this case we are interested in three resources:
+
+* [Mesh](../v1/github.com/solo-io/supergloo/api/v1/mesh.proto.sk.md)
+* [Install](../v1/github.com/solo-io/supergloo/api/v1/install.proto.sk.md)
+* [Pod](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#pod-v1-core) (kubernetes resource)
+
+More information about these resources can be found by clicking the links above. 
+
+At it's core mesh discovery monitors these resources and takes actions based on certain heuristics. These heuristics
```
Comment (ilackarms) [inline supergloo/docs/mesh/mesh-discovery.md]: its

File: supergloo/docs/mesh/mesh-discovery.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+* [Mesh](../v1/github.com/solo-io/supergloo/api/v1/mesh.proto.sk.md)
+* [Install](../v1/github.com/solo-io/supergloo/api/v1/install.proto.sk.md)
+* [Pod](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#pod-v1-core) (kubernetes resource)
+
+More information about these resources can be found by clicking the links above. 
+
+At it's core mesh discovery monitors these resources and takes actions based on certain heuristics. These heuristics 
+vary slightly by the type of mesh we are trying to discover, but the concept is similar. For example, in the case of istio: 
+mesh discovery watches for the existence of istio-pilot pods and creates mesh crds based on the internals of those pods.
```
Comment (ilackarms) [inline supergloo/docs/mesh/mesh-discovery.md]: for the existence of a deployment named `istio-pilot` and discovers the deployed Istio version based on Pilot's image tag

File: supergloo/docs/mesh/mesh-discovery.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+The heuristics we chose are based on our experience working with multiple service meshes.
```
Comment (ilackarms) [inline supergloo/docs/mesh/mesh-discovery.md]: I would remove this line

File: supergloo/docs/mesh/mesh-discovery.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+#### Advanced Config
```
Comment (ilackarms) [inline supergloo/docs/mesh/mesh-discovery.md]: Discovering Mesh Configuration

File: supergloo/docs/mesh/mesh-discovery.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+In order to observe/discover more advanced information, mesh discovery requires the ability to monitor custom resources (CRDs)
```
Comment (ilackarms) [inline supergloo/docs/mesh/mesh-discovery.md]: `In order to observe/discover the existing mesh configuration`

File: supergloo/docs/mesh/mesh-discovery.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+        sg1[Install Event Loop]
+    end
+{{< /mermaid >}}
+
+The above diagram is an approximation of this system working in practice. The lines do not represent order exactly as all of the parts 
+of the system are running concurrently, but it still gives an idea of how the different pieces are created and consumed.
+
+## Mesh Discovery In Practice
+
+As stated above mesh discovery currently requires supergloo to run, so in order to test out mesh discovery we must first install supergloo.
```
Comment (ilackarms) [inline supergloo/docs/mesh/mesh-discovery.md]: `SuperGloo`

File: supergloo/docs/mesh/mesh-discovery.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+To install superloo refer to the previous tutorial on [installation](../install). Once the supergloo cli is installed and supergloo is 
+running we are ready to begin.
+
+If istio is already installed on your system feel free to skip this next step.
```
Comment (ilackarms) [inline supergloo/docs/mesh/mesh-discovery.md]: `s/istio/Istio`

File: supergloo/docs/mesh/mesh-discovery.md [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+* [Mesh](../../v1/github.com/solo-io/supergloo/api/v1/mesh.proto.sk)
+* [Install](../../v1/github.com/solo-io/supergloo/api/v1/install.proto.sk)
+* [Pod](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#pod-v1-core) (kubernetes resource)
+
+More information about these resources can be found by clicking the links above. 
+
+At its core mesh discovery monitors these resources and takes actions based on certain heuristics. These heuristics 
+vary slightly by the type of mesh we are trying to discover, but the concept is similar. For example, in the case of istio: 
+mesh discovery watches for the existence of a deployment named istio-pilot and discovers the deployed Istio version based 
+on Pilot's image tag.
```
Comment (christian-posta) [inline supergloo/docs/mesh/mesh-discovery.md]: So let's try be consistent with the words used. Let's use "SuperGloo" to refer to SuperGloo and "Kubernetes" and Istio, etc. Some are capitalized, some are not, etc .. let's update them throughout the doc to be consistent. 

Also, solo.io should be Solo.io

File: supergloo/docs/mesh/mesh-discovery.md [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+    meshVersion: 1.0.6
+  istio:
+    installationNamespace: istio-system
+    istioVersion: 1.0.6
+status:
+  reported_by: istio-config-reporter
+  state: 1
+```
+
+As you can see supergloo figured out the localtion and version of istio, and now we can go ahead and apply supergloo rules to our mesh.
```
Comment (christian-posta) [inline supergloo/docs/mesh/mesh-discovery.md]: Is that all it does? Can we be more forward looking with what we expect it to do and the overall power this discovery process will bring to SuperGloo (and possibly other projects)?

**Pull request merged**